### PR TITLE
Updated OSM Example Usage Guide to OSM Demo Guide

### PR DESCRIPTION
**Pull request by phillipgibson**
Updated the OSM Example Usage Guide to the OSM Demo Guide. Updated the guide to provide more details on missing installation instructions of prerequisites as well as adding additional context to tasks for completing the demo. Will ultimately want to move this to the landing page for demo instructions.

### CODE REVIEW

File: docs/example/README.md [hunk 1]
```diff
@@ -1,24 +1,38 @@
-# OSM Example Usage Guide
+# OSM Demo Guide
```
Comment (shashankram) [inline docs/example/README.md:1]: This is actually under `docs/examples` and seems to be an example guide. The demo guide is at `demo/README`. Should we rename this to `Manual Demo` guide?

File: docs/example/README.md [hunk 2]
```diff
@@ -1,24 +1,38 @@
...
+  - [Create the Bookstore Applications Namespaces](#create-the-bookstore-application-namespaces)
+  - [Onboard the Namespaces to the OSM Mesh](#onboard-the-namespaces-to-the-osm-mesh)
+  - [Deploy the Bookstore Application](#deploy-the-bookstore-application)
+  - [Checkpoint: What got installed?](#checkpoint-what-got-installed)
+  - [View the Applications UIs](#view-the-application-uis)
+- [Deploy SMI Access Control Policies for OSM](#deploy-smi-access-control-policies-for-osm)
+  - [Allowing the Bookthief Application to access the Mesh](#allowing-the-bookthief-application-to-access-the-mesh)
 - [Traffic Encryption](#traffic-encryption)
-- [Traffic Split](#traffic-split-configuration)
+- [Configure Traffic Split between Two Services](#configure-traffic-split-between-two-services)
```
Comment (shashankram) [inline docs/example/README.md:17]:
```suggestion
- [Configure Traffic Split between Services](#configure-traffic-split-between-two-services)
```

File: docs/example/README.md [hunk 3]
```diff
@@ -1,24 +1,38 @@
...
   - [Deploy v2 of Bookstore](#deploy-v2-of-bookstore)
   - [Update Traffic Split](#update-traffic-split)
-- [Inspect Dashboards](#inspect-dashboards)
+- [Inspect Dashbaords](#inspect-dashboards)
+
+## Overview
+The OSM Demo Guide is designed to quickly allow you to demo and experience the OSM mesh.
```
Comment (shashankram) [inline docs/example/README.md:23]: Should reword this to `manual demo` to avoid confusion with the automated demo.

File: docs/example/README.md [hunk 4]
```diff
@@ -1,24 +1,38 @@
...
+- Kubernetes cluster running Kubernetes v1.15.0 or greater
+- Have `kubectl` CLI installed - [Install and Set Up Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+- kubectl current context is configured for the target cluster install
+  - ```kubectl config current-context```
+- Have GNU Parallel installed
+  - ```wget http://ftp.gnu.org/gnu/parallel/parallel-latest.tar.bz2```
+  - ```tar xjf parallel-latest.tar.bz2```
+  - ```cd parallel-20200722```
+  - ```./configure && make```
+  - ```sudo make install```
```
Comment (shashankram) [inline docs/example/README.md:35]: These instructions are OS platform specific and should be removed or added per platform.

File: docs/example/README.md [hunk 5]
```diff
@@ -28,65 +42,142 @@ Use the [installation guide](/docs/installation_guide.md) to install the `osm` c
...
 The `Bookstore`, `Bookbuyer`, `Bookthief`, `Bookwarehouse` demo applications will be installed in their respective Kubernetes Namespaces. In order for these applications to be injected with a Envoy sidecar automatically, we must add the Namespaces to be monitored by the mesh.
 
+### Create the Bookstore Application Namespaces
 ```bash
-kubectl create ns bookstore
-kubectl create ns bookbuyer
-kubectl create ns bookthief
-kubectl create ns bookwarehouse
-osm namespace add bookstore bookbuyer bookthief bookwarehouse
+for i in bookstore bookbuyer bookthief bookwarehouse; do kubectl create ns $i; done
```
Comment (shashankram) [inline docs/example/README.md:50]: Please revert to the previous, this looks harder to read and is fine to have in the automated script but not as instructions. We don't want users to be understanding how bash for loop syntax should be interpreted.

File: docs/example/README.md [hunk 6]
```diff
@@ -28,65 +42,142 @@ Use the [installation guide](/docs/installation_guide.md) to install the `osm` c
...
 ```
-
+### Onboard the Namespaces to the OSM Mesh
+```bash
+for i in bookstore bookbuyer bookthief bookwarehouse; do osm namespace add $i; done
+```
```
Comment (shashankram) [inline docs/example/README.md:55]: Please revert this.

File: docs/example/README.md [hunk 7]
```diff
@@ -28,65 +42,142 @@ Use the [installation guide](/docs/installation_guide.md) to install the `osm` c
...
 
 *Note: At the moment, you must configure a TrafficSplit object to get your applications set up correctly for inbound traffic because it helps us properly configure the dataplane. We're working on removing the need for this entirely.* [#1370](https://github.com/openservicemesh/osm/issues/1370)
 
-### View Application UIs
+### View the Application UIs
+We will now setup client port forwarding, so we can access the services in the Kubernetes cluster. It is best to start a new terminal session for running the port forwarding script to maintain the port forwarding session, while using the original terminal to continue to issue commands. The port-forward-all.sh script will look for a ".env" file for variables. We will use the reference .env.examples file and then run the port forwarding script. 
+
+In a new terminal session, run the following commands to enable port forwarding into the Kubernetes cluster.
 ```bash
+cp .env.example .env
```
Comment (shashankram) [inline docs/example/README.md:86]: This line should reference why the `.env` is needed (demo/README.md)

File: docs/example/README.md [hunk 8]
```diff
@@ -28,65 +42,142 @@ Use the [installation guide](/docs/installation_guide.md) to install the `osm` c
...
+- http://localhost:8081 - **bookstore-v1**
+- http://localhost:8082 - **bookstore-v2** 
+  - *Note: This page will not be available at this time in the demo. This will become available during the Traffic Split Configuration*
+- http://localhost:8083 - **bookthief**
 
-Position the windows so that you can see all four at the same time. The header at the top of the webpage indicates the application and version. The window for `bookstore-v2` should not be working. Keep it open anyway for the Traffic Split Configuration section later in this document.
+Position the windows so that you can see all four at the same time. The header at the top of the webpage indicates the application and version. 
 
-## Access Control Policies
+## Deploy SMI Access Control Policies for OSM
```
Comment (shashankram) [inline docs/example/README.md:99]: for OSM is implicit?

File: docs/example/README.md [hunk 9]
```diff
@@ -28,65 +42,142 @@ Use the [installation guide](/docs/installation_guide.md) to install the `osm` c
...
+    - books-bought
+  sources:
+  - kind: ServiceAccount
+    name: bookbuyer
+    namespace: bookbuyer
+  - kind: ServiceAccount
+    name: bookthief
+    namespace: bookthief
+ ```
+
```
Comment (shashankram) [inline docs/example/README.md:164]: Can we keep this in the manifests file and add comments there? Looks a bit out of place.

File: docs/example/README.md (hunk 1 above)
Comment (phillipgibson) [reply docs/example/README.md:1]: Yes we need to consolidate all demos into one source of truth on the landing page. We will ultimately remove this file. Once this get's merged I'll start working on updating the landing page with the content.;

Comment (shashankram) [reply docs/example/README.md:1]: We have 2 demos, automated and manual. Both demo guides could co-exist. Did you have something else in mind?

File: docs/example/README.md (hunk 5 above)
Comment (phillipgibson) [reply docs/example/README.md:50]: I want to make the experience as quick and easy as possible. This is a simple bash for loop. I would expect someone looking to run this to be comfortable with this.

File: docs/example/README.md (hunk 6 above)
Comment (phillipgibson) [reply docs/example/README.md:55]: Ref previous comment.

File: docs/example/README.md (hunk 9 above)
Comment (phillipgibson) [reply docs/example/README.md:164]: The goal is for user new to the setup to understand what exactly is happening as they run the manual demo.

File: docs/example/README.md (hunk 5 above)
Comment (shashankram) [reply docs/example/README.md:50]: Its definitely simple for people familiar with bash, but should we assume that in the manual demo guide?

File: docs/example/README.md (hunk 9 above)
Comment (shashankram) [reply docs/example/README.md:164]: Yes but if the manifests are consolidated, might as well keep it there and add comments in the manifests file. We should definitely be adding comments explaining those files.

File: docs/example/README.md (hunk 1 above)
Comment (phillipgibson) [reply docs/example/README.md:1]: This will take over for the manual demo.

File: docs/example/README.md (hunk 2 above)
Comment (phillipgibson) [reply docs/example/README.md:17]: Looks good.

File: docs/example/README.md (hunk 4 above)
Comment (phillipgibson) [reply docs/example/README.md:35]: The use of GNU Parallel is only on Linux. Windows uses will use WSL/WSL2.

File: docs/example/README.md (hunk 1 above)
Comment (shashankram) [reply docs/example/README.md:1]: Got it, thanks.

File: docs/example/README.md (hunk 7 above)
Comment (phillipgibson) [reply docs/example/README.md:86]: Updated more context for the .env file requirement.

File: docs/example/README.md (hunk 9 above)
Comment (phillipgibson) [reply docs/example/README.md:164]: I'm fine with adding the comments to the manifest, but for going through the demo, most people are not going to de-rail from the setup just to read what's in the manifest. They will just create/apply it and move on to the next step. This is just a visual representation of the file change so the user knows they did it correctly.

**Comment by shashankram** [comment]
@phillipgibson this change has a conflict with main, looks good otherwise.
